src/API/Site/Controllers/RestAPI/AuthController.php

```diff
@@ -0,0 +1,133 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\RestAPI;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\WP\OAuthService;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Exception;
+use WP_REST_Request as Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AuthController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\RestAPI
+ */
+class AuthController extends BaseController {
+
+	/**
+	 * @var OAuthService
+	 */
+	protected $oauth_service;
+
+	/**
+	 * Mapping between the client page name and its path.
+	 * The first value is also used as a default,
+	 * and changing the order of keys/values may affect things below.
+	 *
+	 * @var string[]
+	 */
+	private const NEXT_PATH_MAPPING = [
+		'setup-mc'  => '/google/setup-mc',
+		'reconnect' => '/google/settings',
```

Comment:
What do you think if we change `reconnect` with `settings`? Mainly because the query param is `next_page_name` so I think it makes sense to call it 'settings'.

Reply:
Yeah, I think changing to `settings` sounds more straightforward to me. I updated in b41bad3, however in Google/Jetpack's AccountController they still use reconnect as well as some places in the frontend, I will create a follow up issue so we could do it in another PR.